packages/cli/test/unit/commands/dev/index.test.ts

```diff
@@ -1,9 +1,147 @@
-import { describe } from 'vitest';
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+import dev from '../../../../src/commands/dev';
+import { client } from '../../../mocks/client';
+import { type fs, vol } from 'memfs';
+import { useUser } from '../../../mocks/user';
+import { useTeams } from '../../../mocks/team';
+import { useProject } from '../../../mocks/project';
+
+vi.mock(import('../../../../src/util/dev/server'), () => {
+  class DevServer {
+    devCommand = 'framework dev';
+    feed() {}
+    stop() {}
+    start() {}
+  }
+  return { default: DevServer };
+});
+
+vi.mock('node:fs/promises', async () => {
+  const memfs: { fs: typeof fs } = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+vi.mock('node:fs', async () => {
+  const memfs: { fs: typeof fs } = await vi.importActual('memfs');
+  return memfs;
+});
+
+afterEach(() => {
+  // __VERCEL_DEV_RUNNING is set when `vercel dev` runs to act as a lock.
+  // It's unset as a side effect of  the process exiting. This won't work under test
+  // where `vercel dev` can be invoked several times in a row.
+  vi.stubEnv('__VERCEL_DEV_RUNNING', undefined);
+  vol.reset();
+});
 
 describe('dev', () => {
-  describe.todo('[dir]');
-  describe.todo('--listen');
-  describe.todo('--yes');
-  describe.todo('--port');
-  describe.todo('--confirm');
+  const projectId = 'prj_whatever123';
+  const orgId = 'team_123';
+  const projectName = 'project-name';
+  const projectPath = `/user/name/code/${projectName}`;
+
+  beforeEach(() => {
+    useUser();
+    useTeams(orgId);
+    useProject({
+      id: projectId,
+      name: projectName,
+    });
+
+    const json = {
+      '.vercel/project.json': JSON.stringify({
+        projectId,
+        orgId,
+      }),
+    };
+    vol.fromJSON(json, projectPath);
+  });
+
+  describe('[dir]', () => {
+    it('tracks the dir if supplied', async () => {
+      client.setArgv('dev', projectPath);
+      const exitCodePromise = dev(client);
+
+      // dev is an odd duck in that normally only exists on SIGTERM
+      await expect(exitCodePromise).resolves.toEqual(undefined);
+      expect(client.telemetryEventStore).toHaveTelemetryEvents([
+        {
+          key: 'argument:dir',
+          value: '[REDACTED]',
+        },
+      ]);
+    });
+  });
+
+  describe('--listen', () => {
+    it('tracks the listen option if supplied', async () => {
+      vi.spyOn(process, 'cwd').mockReturnValue(projectPath);
+
+      client.setArgv('dev', '--listen=9090');
+      const exitCodePromise = dev(client);
+
+      // dev is an odd duck in that normally only exists on SIGTERM
```

Comment:
```suggestion
      // dev is an odd duck in that normally only exits on SIGTERM
```
This same comment is found below as well.

Reply:
@jeffsee55 can I fix this in a follwup? if tests go green? I've been fighting CI all day and this will be my 11th run